lib/algolia/index.rb

```diff
@@ -409,7 +409,9 @@ def delete_by_query(query, params = nil)
       loop do
         res = search(query, params)
         break if res['hits'].empty?
-        res = delete_objects(res['hits'].map { |h| h['objectID'] })
+        ids = res['hits'].map { |h| h['objectID'] }
+        res = delete_objects(ids)
+        break if ids.size != 1000 # no need to wait_task if we had less than 1000 objects matching; we won't get more after anyway.
```

Comment:
@raphi and @redox, I've played around with this in my own project and had very successful results, I tried deleting ~3000 objects with it and it happened within milliseconds instead of more than 5 minutes. I've opened PR #211 with the solution that worked for me based on @raphi's suggestion.